features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/textcomposer/MessageComposerView.kt

```diff
@@ -53,13 +70,42 @@ fun MessageComposerView(
         composerMode = state.mode,
         onCloseSpecialMode = ::onCloseSpecialMode,
         onComposerTextChange = ::onComposerTextChange,
+        onAddAttachment = {
+            cameraLauncher.launch()
+        },
         composerCanSendMessage = state.isSendButtonVisible,
         composerText = state.text?.charSequence?.toString(),
         isInDarkMode = !ElementTheme.colors.isLight,
         modifier = modifier
     )
 }
 
+@Composable
+internal fun CameraPhotoPicker(onResult: (Uri?) -> Unit): PickerLauncher<Uri, Boolean> {
+    // Screenshot tests and preview can't handle FileProviders, so we might as well disable the whole picker
+    return if (LocalInspectionMode.current) {
+        NoOpPickerLauncher()
+    } else {
+        val context = LocalContext.current
+        val tmpFile = remember {
+            val filename = UUID.randomUUID().toString()
+            File(context.cacheDir, filename)
+        }
+        val tmpFileUri = remember(tmpFile) {
+            val authority = "${context.packageName}.fileprovider"
+            FileProvider.getUriForFile(context, authority, tmpFile)
+        }
```

Comment:
These lines could probably be extracted to a helper function so they can be used for both camera photo and video pickers, Also, I'm not sure if we might want to use a filename based on a constant instead of using a hash here.